pkg/state/invoke_applier.go

```diff
@@ -432,12 +442,22 @@ func (ia *invokeApplier) fallibleValidation(tx proto.Transaction, info *addlInvo
 			}
 		case *proto.AttachedPaymentScriptAction:
 			// Perform transfers.
-			recipientAddress := a.Recipient.Address()
-			totalChanges.appendAddr(*recipientAddress)
 			assetExists := ia.stor.assets.newestAssetExists(a.Asset)
-			if !assetExists {
+			if !assetExists { // check asset existence
 				return proto.DAppError, info.failedChanges, errors.New("invalid asset in transfer")
 			}
+			recipientAddress, rErr := recipientToAddress(a.Recipient, ia.stor.aliases)
```

Comment:
I think the same works for `AttachedPaymentScriptAction` https://github.com/wavesplatform/gowaves/blob/66b44f8b076f7e8c42a08ad66dff07c355b81fc1/pkg/ride/environment.go#L796